web/src/features/time/BottomSheet.ts

```diff
@@ -0,0 +1,340 @@
+export type SnapPoint = number;
+
+interface BottomSheetEvents {
+  onOpen?: () => void;
+  onClose?: () => void;
+  onSnap?: (snapIndex: number) => void;
+}
+
+export interface BottomSheetProps extends BottomSheetEvents {
+  snapPoints?: SnapPoint[];
+  backgroundColor?: string;
+  excludeElement?: HTMLElement;
+}
+
+function createBackdropElement(bgColor: string): HTMLDivElement {
+  const backdrop = document.createElement('div');
+  const hasTailwindBackground = bgColor.includes('bg-');
+
+  if (!hasTailwindBackground) {
+    Object.assign(backdrop.style, {
+      position: 'absolute',
+      left: '0',
+      right: '0',
+      bottom: '0',
+      backgroundColor: bgColor,
+      transition: 'height 0.3s ease-out',
+    });
+  }
+  if (hasTailwindBackground) {
+    Object.assign(backdrop.style, {
+      position: 'absolute',
+      left: '0',
+      right: '0',
+      bottom: '0',
+      transition: 'height 0.3s ease-out',
+    });
+    backdrop.classList.add(bgColor);
+  }
+
+  return backdrop;
+}
+
+function getSafeAreaBottom(): number {
+  const windowHeight = window.innerHeight;
+  const documentHeight = document.documentElement.clientHeight;
+  return windowHeight - documentHeight;
+}
+
+function isExcludedElement(props: BottomSheetProps, target: EventTarget | null): boolean {
+  if (!props.excludeElement || !target) {
+    return false;
+  }
+  return props.excludeElement.contains(target as Node);
+}
+
+export function createBottomSheet(element: HTMLElement, props: BottomSheetProps) {
+  const { backgroundColor = 'white' } = props;
+
+  const state = {
+    currentSnap: 0,
+    childrenHeight: 0,
+    snapPointsWithChildHeight: [] as number[],
+  };
+
+  const backdropElement = createBackdropElement(backgroundColor);
+  const childrenElement = element.children[0] as HTMLElement;
+
+  function init(): () => void {
+    setupElementStyles(element);
+    element.append(backdropElement);
+    updateChildrenHeight();
+    setSnap(0);
+    setupAccessibility();
+    return addEventListeners();
+  }
+
+  function setupAccessibility(): void {
+    element.setAttribute('role', 'dialog');
```

Comment:
Instead of setting a ARIA attribute we should use the `<dialog>` element:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog